### ISSUE TITLE
0.7.0 breaks translates without options

### ISSUE THREAD

**Issue by pwim** [shioyama/mobility#243]
``` ruby
class Question < ApplicationRecord
  translates :text
end
```
results in a syntax error under 0.7.0:
```
  /usr/local/var/rbenv/versions/2.5.1/lib/ruby/gems/2.5.0/gems/mobility-0.7.0/lib/mobility/plugins/locale_accessors.rb:60: syntax error, unexpected '{', expecting ';' or '\n'
          def name_{:en=>:ja, :ja=>:en}(options =...
                   ^
```
Switching to having an empty options works around the bug.
``` ruby
class Question < ApplicationRecord
  translates :text, {}
end
```
I noticed all the test cases have an explicit options hash, but I'm assuming this isn't required.

**Comment by glebm** [comment]
Same with 0.7.2

**Comment by shioyama** [comment]
Sorry, somehow missed this issue. I'll fix it today.

**Comment by shioyama** [comment]
Hmm strange, from the error above `I18n.available_locales` seems to be returning the fallbacks, but I don't see this on a dummy app. I'll have to dig further.

**Comment by shioyama** [comment]
Ok I created a clean dummy app with a single model `Question`, with:

```ruby
class ApplicationRecord < ActiveRecord::Base
  self.abstract_class = true
  extend Mobility
end
```

```ruby
class Question < ApplicationRecord
  translates :text
end
```

And I can't reproduce this... model loads fine and I can instantiate and create a question with a `text` attribute value. Must be something in your config that is triggering it.

**Comment by glebm** [comment]
It looks like this is caused by Mobility-enabled readers getting defined before `I18n` is initialized by Rails:

```console
> I18n.available_locales
[:en, :ru, :ro, :"zh-TW", :id, :tr, :hu, :"zh-CN", :it, :nl, :de, :da, :es, :ja, :fr, :cs, :"pt-BR", :sk, :"en-GB", :"fi-FI", :"en-US", :pl, :"en-PAK", :ca, :bg, :ko, :"en-NG", :uk, :"nb-NO", :"en-SG", :"ca-CAT", :"en-ZA", :"de-AT", :"en-UG", :"en-IND", :pt, :"da-DK", :vi, :"en-BORK", :"en-CA", :"en-au-ocker", :he, :"en-AU", :sv, :fa, :"es-MX", :nep, :"de-CH", :"en-MS", :"en-NZ"]

> Rails.application.config.i18n.available_locales
["en", "es"]
```

Note that Mobility uses `I18n.available_locales`:

https://github.com/shioyama/mobility/blob/da31515a7137535f2e54e8632a68e23d741683b4/lib/mobility/configuration.rb#L107

One of the `available_locales` is `:"en-au-ocker"` ([from faker](https://github.com/stympy/faker/blob/master/lib/locales/en-au-ocker.yml) in my case), which cannot be normalized by Mobility.

Rails initializes `I18n.available_locales` *after* the initializers have run:

https://github.com/rails/rails/blob/v5.2.0/activesupport/lib/active_support/i18n_railtie.rb#L15-L19

Note that this bug also existed in mobility 0.6.0, however methods were defined with `define_method` which allows almost any method name.

A workaround that worked for me is:

```ruby
Mobility.configure do |config|
  config.default_options[:locale_accessors] = Rails.application.config.i18n.available_locales
end
```

Perhaps mobility should default to `Rails.application.configuration.available_locales` if in Rails?

**Comment by shioyama** [comment]
> which cannot be normalized by Mobility, causing a syntax error.

Oh, I see:

```ruby
Mobility.normalize_locale(:"en-au-ocker")
#=> "en_au-ocker"
```

I actually thought `sub` would convert all dashes to underscores here. So I consider that a bug, easily fixable.

So that's one thing. But the other one is load order and `I18n.available_locales`. I'll have to dig into that one a bit more, but I'm confused why Mobility would define the readers before Rails initializes I18n.

**Comment by shioyama** [comment]
> It looks like this was caused for me by Mobility readers getting defined before I18n is initialized by Rails

I can't reproduce this. Mobility only pulls `I18n.available_locales` at the moment when it is defining the model, by which time Rails should have done whatever configuration it needs to do, right?

When I set a different `available_locales` on `Rails.application.config.i18n`, then pry into the model definition, I find that `I18n.available_locales` is the same value. So I'm not sure why in your case it the two values seem not to be synced.

**Comment by shioyama** [comment]
This is actually exactly why Mobility uses a Proc to get the available locales, see 3cd786814.

**Comment by glebm** [comment]
If the model class is referenced in an initializer, then the proc will get called before I18n.available_locales is set

**Comment by shioyama** [comment]
I merged #254, @glebm please try it on `master`. I'll wait until the next minor to release it.

**Comment by glebm** [comment]
@shioyama master works, thanks!
